File: solutions/java/bird-watcher/1/src/main/java/BirdWatcher.java

class BirdWatcher {
    private final int[] birdsPerDay;

    public BirdWatcher(int[] birdsPerDay) {
        this.birdsPerDay = birdsPerDay.clone();
    }

    public int[] getLastWeek() {
        int[] week={0,2,5,3,7,8,4};
        return week;
    }

    public int getToday() {
        return birdsPerDay[birdsPerDay.length - 1];
    }

    public void incrementTodaysCount() {
        birdsPerDay[birdsPerDay.length - 1]++;
    }

    public boolean hasDayWithoutBirds() {
        for( int day : birdsPerDay){
            if(day==0)
                return true;
        }
        return false;
    }

    public int getCountForFirstDays(int numberOfDays) {
        int sum=0;
        for(int i=0; i<Math.min(numberOfDays, birdsPerDay.length); i++){
            sum += birdsPerDay[i];
        }
        return sum;
    }

    public int getBusyDays() {
        int count=0;
        for(int bird : birdsPerDay){
            count += bird>=5 ? 1 : 0;
        }
        return count;
    }
}

File: solutions/julia/bird-watcher/1/bird-watcher.jl
function today(birds_per_day)
    last(birds_per_day)
end

function increment_todays_count(birds_per_day)
    birds_per_day[end] += 1
    birds_per_day 
end

function has_day_without_birds(birds_per_day)
    (0 in birds_per_day)
end

function count_for_first_days(birds_per_day, num_days)
    sum(birds_per_day[1:num_days])
end

function busy_days(birds_per_day) 
    length(birds_per_day[birds_per_day.>=5])
end

function average_per_day(week1, week2)
    [(week1[i] + week2[i])*0.5 for i in                                     1:min(length(week1),length(week2))]
end
